### PR TITLE
[OTEL-931] Convert pkg/autodiscovery/common/types to go modules

### PR DESCRIPTION
**Pull request by songy23**
### What does this PR do?
Convert pkg/autodiscovery/common/types into go modules.

### Motivation
This is part of a large PR that converts convert metrics serializer in Agent to module: https://github.com/DataDog/datadog-agent/pull/18712, related to Agent modularization

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by chouquette — APPROVED** [review]
LGTM for agent platform files

File: pkg/autodiscovery/common/types/go.mod [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+replace github.com/DataDog/datadog-agent/pkg/util/log => ../../../util/log/
+
+require (
+	github.com/DataDog/datadog-agent/pkg/util/log v0.0.0-00010101000000-000000000000
+	github.com/stretchr/testify v1.8.4
+)
+
+require (
+	github.com/DataDog/datadog-agent/pkg/util/scrubber v0.49.0-rc.2 // indirect
```
Comment (pgimalac) [inline pkg/autodiscovery/common/types/go.mod:13]: Don't you need a replace for `scrubber` too ?

Comment (songy23) [reply pkg/autodiscovery/common/types/go.mod:16]: Good catch!

**Review by dineshg13 — APPROVED** [review]

**Review by clamoriniere — COMMENTED** [review]
could you document which `structs` or `fields`  from auto-discovery is needed? 
And where it is used? 

I would like to understand if we can move them outside the auto-discovery package instead of create yet another sub-module 🙇 
If it is like in our previous discussion

**Comment by liustanley** [comment]
> could you document which `structs` or `fields` from auto-discovery is needed? And where it is used?
> 
> I would like to understand if we can move them outside the auto-discovery package instead of create yet another sub-module 🙇 If it is like in our previous discussion

The only usage of this module is for `PrometheusCheck` in this file: https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config.go#L132

**Comment by clamoriniere** [comment]
> > could you document which `structs` or `fields` from auto-discovery is needed? And where it is used?
> > I would like to understand if we can move them outside the auto-discovery package instead of create yet another sub-module 🙇 If it is like in our previous discussion
> 
> The only usage of this module is for `PrometheusCheck` in this file: https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config.go#L132

This function`PrometheusScrapeChecksTransformer` doesn't seems to be use anywere (except in a unit-test) so we cloud remove it or move it inside the Autodiscovery Package

**Comment by liustanley** [comment]
> > > could you document which `structs` or `fields` from auto-discovery is needed? And where it is used?
> > > I would like to understand if we can move them outside the auto-discovery package instead of create yet another sub-module 🙇 If it is like in our previous discussion
> > 
> > 
> > The only usage of this module is for `PrometheusCheck` in this file: https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config.go#L132
> 
> This function`PrometheusScrapeChecksTransformer` doesn't seems to be use anywere (except in a unit-test) so we cloud remove it or move it inside the Autodiscovery Package

@clamoriniere It's actually used in `pkg/config/config.go` in the function `InitConfig`, which is a function we need to refactor the config setup step into a module: https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config.go#L673. If we were to move `PrometheusScrapeChecksTransformer` into the Autodiscovery package, we would still need to make it into a module.

**Comment by clamoriniere** [comment]
hi @liustanley 

I made a pull request https://github.com/DataDog/datadog-agent/pull/20850 to remove the `pkg/autodiscovert/common/types` dependency from the `config` package.

**Pull request closed without merging**